### PR TITLE
Backport main-next non-clangable

### PR DESCRIPTION
**Pull request by quaresmajose**

### CODE REVIEW

**Comment by quaresmajose** [comment]
should unblock https://github.com/foundriesio/lmp-manifest/pull/277

File: meta-lmp-base/conf/distro/include/non-clangable.inc [hunk 1]
```diff
@@ -73,6 +73,21 @@ OBJCOPY:pn-docker-compose:toolchain-clang = "${HOST_PREFIX}objcopy"
...
+OBJCOPY:pn-linux-lmp-lts:toolchain-clang = "${HOST_PREFIX}objcopy"
+OBJCOPY:pn-linux-lmp-rt:toolchain-clang = "${HOST_PREFIX}objcopy"
+OBJCOPY:pn-linux-lmp-ea-imx:toolchain-clang = "${HOST_PREFIX}objcopy"
+OBJCOPY:pn-linux-lmp-fslc-imx:toolchain-clang = "${HOST_PREFIX}objcopy"
+OBJCOPY:pn-linux-lmp-fslc-imx-rt:toolchain-clang = "${HOST_PREFIX}objcopy"
+OBJCOPY:pn-linux-lmp-rpi:toolchain-clang = "${HOST_PREFIX}objcopy"
+OBJCOPY:pn-linux-lmp-ti-staging:toolchain-clang = "${HOST_PREFIX}objcopy"
+OBJCOPY:pn-linux-lmp-toradex-imx:toolchain-clang = "${HOST_PREFIX}objcopy"
+OBJCOPY:pn-linux-lmp-xlnx:toolchain-clang = "${HOST_PREFIX}objcopy"
+OBJCOPY:pn-linux-tegra:toolchain-clang = "${HOST_PREFIX}objcopy"
```
Comment (ricardosalveti) [inline meta-lmp-base/conf/distro/include/non-clangable.inc:89]: Do you know why this started to happen?

Comment (quaresmajose) [reply meta-lmp-base/conf/distro/include/non-clangable.inc:89]: No, don't know where it is been introduced but can find it locally.
On main next I think see something about this on clang layer but on this stable update this makes less sense.
Anyway it fixes the build https://github.com/foundriesio/lmp-manifest/pull/284 that almost done.

Comment (ricardosalveti) [reply meta-lmp-base/conf/distro/include/non-clangable.inc:89]: Sure, I'm ok merging this, I would just like to have a better understanding so we know when to revert later on.

Comment (quaresmajose) [reply meta-lmp-base/conf/distro/include/non-clangable.inc:89]: Right, I don't even know what happened just to change from one branch to another.
I will find something about this clang issues to update the commit message.

**Review by MrCry0 — APPROVED** [review]
LGTM

File: meta-lmp-base/conf/distro/include/non-clangable.inc (hunk 1 above)
Comment (ricardosalveti) [reply meta-lmp-base/conf/distro/include/non-clangable.inc:89]: This seems that was added in meta-clang as well:

```
+# Subprocess output:mips-yoe-linux-llvm-objcopy: error: Link field value 22 in section .rel.dyn is not a symbol table
+# also seen on riscv64 and x86-64
+OBJCOPY:pn-linux-yocto:toolchain-clang = "${HOST_PREFIX}objcopy"
```

**Comment by ricardosalveti** [comment]
```
commit 299a5fd24b7bee2191127c9fef8bd39d807f3492
Author: Khem Raj <raj.khem@gmail.com>
Date:   Thu Dec 15 10:41:47 2022 -0800

    linux-yocto: Use binutils provided strip

    We are not _yet_ using clang as kernel compiler

    Signed-off-by: Khem Raj <raj.khem@gmail.com>

diff --git a/conf/nonclangable.conf b/conf/nonclangable.conf
index f322fa1..93d5d7e 100644
--- a/conf/nonclangable.conf
+++ b/conf/nonclangable.conf
@@ -418,6 +418,7 @@ RANLIB:append:pn-tcf-agent:toolchain-clang = " $@"
 # Subprocess output:mips-yoe-linux-llvm-objcopy: error: Link field value 22 in section .rel.dyn is not a symbol table
 # also seen on riscv64 and x86-64
 OBJCOPY:pn-linux-yocto:toolchain-clang = "${HOST_PREFIX}objcopy"
+STRIP:pn-linux-yocto:toolchain-clang = "${HOST_PREFIX}strip"

 # see https://github.com/llvm/llvm-project/issues/53948
 OBJCOPY:pn-opensbi:toolchain-clang = "${HOST_PREFIX}objcopy"
```

**Comment by ricardosalveti** [comment]
So yeah, please add the objcopy and strip changes for all kernel recipes we use (since kernel is indeed not yet using clang during build-time).

**Comment by quaresmajose** [comment]
We have now most of the layers updated with https://github.com/foundriesio/lmp-manifest/pull/287 so my next task will be update oe-core layer and find where was introduced this issues that mainly affect the clang toolchain.

**Comment by quaresmajose** [comment]
@MrCry0 we have already meta-clang updated with latest kirkstone version but this clang issues comes when we bump oe-core so the root cause is in oe-core and not in meta-clang imo

**Comment by quaresmajose** [comment]
This are been integrated in https://github.com/foundriesio/lmp-manifest/pull/287

**Pull request closed without merging**